src/python/pants/backend/python/util_rules/lockfile_metadata.py

```diff
@@ -59,18 +59,14 @@ def from_lockfile(
             ),
         )
 
-    def _header_dict(self) -> dict[Any, Any]:
-        """Produce a dictionary to be serialized into the lockfile header.
-
-        Subclasses should call `super` and update the resulting dictionary.
-        """
-
-        d = super()._header_dict()
-        d["valid_for_interpreter_constraints"] = [
-            str(ic) for ic in self.valid_for_interpreter_constraints
-        ]
-
-        return d
+    @classmethod
+    def header_attrs(cls, instance: LockfileMetadata) -> dict[Any, Any]:
```

Comment:
How about calling it `additional_header_attrs`? That way we make more clear this is _augmenting_ whatever the superclasses are doing, not overwriting.